app/code/core/Mage/Core/Model/Abstract.php

```diff
@@ -127,7 +127,7 @@ protected function _setResourceModel($resourceName, $resourceCollectionName = nu
     /**
      * Get resource instance
      *
-     * @return Mage_Core_Model_Resource_Db_Abstract|object|string
+     * @return Mage_Core_Model_Resource_Db_Abstract|object
```

Comment:
It can return `false`.

```suggestion
     * @return Mage_Core_Model_Resource_Db_Abstract|object|false
```

Reply:
You`re right, `false` is a possible ....

How about ...

```
        if (empty($this->_resourceName)) {
            Mage::throwException(Mage::helper('core')->__('Resource is not set.'));
        }

        // changed from here
        $resource = Mage::getResourceSingleton($this->_resourceName);
        if ($resource !== false) {
            Mage::throwException(Mage::helper('core')->__('Resource is not found.'));
        }

        return $resource;
```